### PR TITLE
feat(ui): Implement `<PhoneNumberField />` component

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

- Implements `<PhoneNumberField />`
- Pulls data and hook from clerk-js https://github.com/clerk/javascript/blob/49e7886f1f24a8ac95514e8fe8b0006dca715a79/packages/clerk-js/src/ui/elements/PhoneInput/useFormattedPhoneNumber.ts#L17

https://github.com/clerk/javascript/assets/825855/931db79e-14c2-4fc1-8ed8-f9108604ce85

https://linear.app/clerk/issue/SDKI-10/build-phone-country-code-combobox

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/ui/src/common/PhoneNumberInput/index.tsx
Comment (joe-bell) [inline packages/ui/src/common/PhoneNumberInput/index.tsx:1]: Perhaps a nit but does it make more sense to call this `PhoneNumberField` and include the `label` prop?

Comment (alexcarpenter) [reply packages/ui/src/common/PhoneNumberInput/index.tsx:1]: Thinking about how we'd swap between email and phone number like this is why I built the input on its own for now. Likely multiple components that make use of the field internally. e.g. `<PhoneNumberField />` `<EmailOrPhoneNumberField />`?


https://github.com/clerk/javascript/assets/825855/7cbd03f8-9a11-44ad-850a-ee9ffcbb8f2c

**Comment by joe-bell** [comment]
Testing locally it seems that the prefix gets added; but my assumption is the dropdown will take care of that already?
CleanShot 2024-06-06 at 16 17 43@2x

File: packages/ui/src/common/PhoneNumberInput/index.tsx
Comment (joe-bell) [reply packages/ui/src/common/PhoneNumberInput/index.tsx:1]: Hmmm. What if:

1. We had a field for each
2. We handle how to display said fields in a separate component; which we wrap in `relative` class and absolutely position the specific caption

What do you reckon?

Comment (joe-bell) [reply packages/ui/src/common/PhoneNumberInput/index.tsx:1]: Gotcha, in that case I think let's add a `alternative` slot or something like you suggested; but we create a separate component to handle the logic of which one to display?

Comment (alexcarpenter) [reply packages/ui/src/common/PhoneNumberInput/index.tsx:1]: Sounds good, I will wrap in a field and label.

**Comment by joe-bell** [comment]
Sorry Alex, got my PRs muddled 😅

**Comment by alexcarpenter** [comment]
> Testing locally it seems that the prefix gets added; but my assumption is the dropdown will take care of that already? CleanShot 2024-06-06 at 16 17 43@2x

As you add more numbers, it removes the plus to my knowledge.

**Comment by alexcarpenter** [comment]
> Testing locally it seems that the prefix gets added; but my assumption is the dropdown will take care of that already? CleanShot 2024-06-06 at 16 17 43@2x

The jump from 2-3 in gray scale is fairly drastic.

## `gray-2`
Screenshot 2024-06-06 at 1 44 14 PM


## `gray-3`
Screenshot 2024-06-06 at 1 44 33 PM

**Comment by joe-bell** [comment]
Ooof I agree this is tough, I think we really need a gray in the middle here 🤔 

Anyway, let's loop back. Otherwise this looks great!

**Review by joe-bell — APPROVED** [review]

**Pull request merged**